tools/cli/internal/openapi/versions.go

```diff
@@ -18,13 +18,24 @@ import (
 
 	"github.com/getkin/kin-openapi/openapi3"
 	"github.com/mongodb/openapi/tools/cli/internal/apiversion"
+	"github.com/mongodb/openapi/tools/cli/internal/openapi/filter"
 )
 
 // ExtractVersions extracts version strings from an OpenAPI specification.
-func ExtractVersions(oas *openapi3.T) []string {
-	versions := make(map[string]struct{})
+func ExtractVersions(oas *openapi3.T, env string) ([]string, error) {
+	// We need to remove the version that are hidden for the given environment
+	doc, err := filter.ApplyFiltersWithInit(oas, filter.NewMetadata(nil, env), func(oas *openapi3.T, metadata *filter.Metadata) []filter.Filter {
+		return []filter.Filter{
+			filter.InitHiddenEnvsFilter(oas, metadata),
+		}
+	})
```

Comment:
Applying the hidden env filter when retrieving the versions for a spec

Reply:
if no env is provided as a flag we should return all versions in the v2 file

Reply:
the core of the command is to find versioned matching content in the spec. if an env is provided, then we apply filtering

Reply:
Updated. Thanks!